extendedcopy_test.go

```diff
@@ -889,6 +1003,54 @@ func TestExtendedCopyGraph_FilterArtifactTypeWithRegex(t *testing.T) {
 	verifyCopy(dst, copiedIndice, uncopiedIndice)
 }
 
+func TestExtendedCopyGraph_FilterArtifactTypeWithNoRegex(t *testing.T) {
+	// generate test content
+	var blobs [][]byte
+	var descs []ocispec.Descriptor
+	appendBlob := func(mediaType string, blob []byte) {
+		blobs = append(blobs, blob)
+		descs = append(descs, ocispec.Descriptor{
+			MediaType: mediaType,
+			Digest:    digest.FromBytes(blob),
+			Size:      int64(len(blob)),
+		})
+	}
+	generateArtifactManifest := func(subject ocispec.Descriptor, artifactType string) {
+		var manifest artifactspec.Manifest
+		artifactSubject := descriptor.OCIToArtifact(subject)
+		manifest.Subject = &artifactSubject
+		manifest.ArtifactType = artifactType
+		manifestJSON, err := json.Marshal(manifest)
+		if err != nil {
+			t.Fatal(err)
+		}
+		appendBlob(artifactspec.MediaTypeArtifactManifest, manifestJSON)
+	}
+
+	appendBlob(ocispec.MediaTypeImageConfig, []byte("foo")) // descs[0]
+	generateArtifactManifest(descs[0], "good-bar-yellow")   // descs[1]
+	generateArtifactManifest(descs[0], "bad-woo-red")       // descs[2]
+	generateArtifactManifest(descs[0], "bad-bar-blue")      // descs[3]
+	generateArtifactManifest(descs[0], "bad-bar-red")       // descs[4]
+	generateArtifactManifest(descs[0], "good-woo-pink")     // descs[5]
+
+	ctx := context.Background()
+
+	src := memory.New()
+	for i := range blobs {
+		err := src.Push(ctx, descs[i], bytes.NewReader(blobs[i]))
+		if err != nil {
+			t.Errorf("failed to push test content to src: %d: %v", i, err)
+		}
+	}
+
+	opts := oras.ExtendedCopyGraphOptions{}
+	opts.FilterArtifactType(nil)
```

Comment:
I think we can also test the cases of calling `FilterArtifactType` and `FilterAnnotation` **multiple times**, where one or some of the calls have nil arguments.